diff --git a/CMakeLists.txt b/CMakeLists.txt
index bfaf80a5e..6a3d03ca3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -345,7 +345,7 @@ endif()
 
 # MIPP
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/lib/MIPP/src/mipp.h")
-    aff3ct_target_compile_definitions(PUBLIC MIPP_ENABLE_BACKTRACE)
+    #aff3ct_target_compile_definitions(PUBLIC MIPP_ENABLE_BACKTRACE)
     aff3ct_target_include_directories(PUBLIC "lib/MIPP/src" "MIPP")
     message(STATUS "AFF3CT - Header found: MIPP")
 else()
diff --git a/src/Tools/system_functions.cpp b/src/Tools/system_functions.cpp
index 942338569..32785b140 100644
--- a/src/Tools/system_functions.cpp
+++ b/src/Tools/system_functions.cpp
@@ -1,4 +1,4 @@
-#if (defined(__GNUC__) || defined(__clang__) || defined(__llvm__)) && (defined(__linux__) || defined(__linux) || defined(__APPLE__))
+#if ((defined(__GNUC__) || defined(__clang__) || defined(__llvm__)) && (defined(__linux__) || defined(__linux) || defined(__APPLE__)) && defined(ENABLE_BACKTRACE))
 #include <execinfo.h> // backtrace, backtrace_symbols
 #include <cxxabi.h>   // __cxa_demangle
 #endif
@@ -26,7 +26,7 @@ std::string aff3ct::tools::get_back_trace(int first_call)
 {
 	std::string bt_str;
 
-#if (defined(__GNUC__) || defined(__clang__) || defined(__llvm__)) && (defined(__linux__) || defined(__linux) || defined(__APPLE__))
+#if (defined(__GNUC__) || defined(__clang__) || defined(__llvm__)) && (defined(__linux__) || defined(__linux) || defined(__APPLE__)) && defined(ENABLE_BACKTRACE)
 	const int bt_max_depth = 32;
 	void *bt_array[bt_max_depth];
 
